protocol/x/vault/keeper/msg_server_deposit_to_megavault.go

```diff
@@ -0,0 +1,47 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	assettypes "github.com/dydxprotocol/v4-chain/protocol/x/assets/types"
+	sendingtypes "github.com/dydxprotocol/v4-chain/protocol/x/sending/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// DepositToMegavault deposits from a subaccount to megavault.
+func (k msgServer) DepositToMegavault(
```

Comment:
nit: For this method to be re-useable within `x/listing` it should be on the keeper and not the message server.

Reply:
Yeah I was going to work on getting deposit to be what the spec says in a separate PR (moving this to keeper + taking in minShares + returning mintedShares)